trio/_core/_run.py

```diff
@@ -1386,26 +1506,29 @@ class _RunStatistics:
 # worker thread.
 @attr.s(eq=False, hash=False, slots=True)
 class GuestState:
-    runner = attr.ib()
-    run_sync_soon_threadsafe = attr.ib()
-    run_sync_soon_not_threadsafe = attr.ib()
-    done_callback = attr.ib()
-    unrolled_run_gen = attr.ib()
-    _value_factory: Callable[[], Value] = lambda: Value(None)
-    unrolled_run_next_send = attr.ib(factory=_value_factory, type=Outcome)
-
-    def guest_tick(self):
+    runner: Runner = attr.ib()
+    run_sync_soon_threadsafe: Callable[[Callable[[], object]], object] = attr.ib()
+    run_sync_soon_not_threadsafe: Callable[[Callable[[], object]], object] = attr.ib()
```

Comment:
I thought that it was `Callable[[Callable[[], None], Any]`

Reply:
That would also work, but since we don't actually use the return type, doing it this way means that it accepts functions that do return some value other than `None`.

Reply:
https://typing.readthedocs.io/en/latest/source/best_practices.html#using-any-and-object
> [...] if the return value of a callback is ignored, annotate it with object